javascript/react-js/react-introduction.md

```diff
@@ -0,0 +1,136 @@
+
+- `App` which represents your main application and which will be the parent of all the other components
+- `Navbar` which will the the Navigation bar
+- `MainArticle` which will be the component that renders your content
+- `NewsletterForm` which is a simple form, that let's a user input his email to receive the weekly Newsletter
+
+In React each component is an own ES6 module, which is per se independent. But through the in ES6 introduced `import` statement you can import components into other components like so:
+`import ExampleComponent from "./components/ExampleComponent`. This allowes us to write each component as an own module in a seperate file, and later on import all modules to the file that contains them. (In our case that would be App.js)
+
+At the beginning it might be a little bit difficult to find out the best component structure, especially when state and props come into play, but that's the topic for the next sections. Don't worry too much about the component structur for now, this comes with experience, you'll figure it out. React components in general can and usually have parent- or child components. This system of structuring your applications component-base helps to keep your code structured and makes it easy to keep track of your components relationships with each other.
+
+To give you an example of a very basic component, it could looks like this:
+
+```javascript
+import React, { Component } from 'react'
+
+class App extends Component {
+    constructor() {
+        super()
+    }
+
+    {/* Javascript functions can we written here */}
+
+    render() {
+        return (
+            <div className="App">
+            Hello World!
+            </div>
+        )
+    }
+}
+
+export default App
+```
+
+Don't get overwhelmed it isn't as difficult as it might look at the beginning. Let's walk through it step by step.
+
+1. With `import React, { Component } from "react"` we are importing React and the Components module from react, which allows us to create a class component
+2. Secondly we are declaring the class component, which really is just a Javascript class, that extends Component, which we imported earlier. One thing to notice is that React components should always be declared with a capital letter at the beginning.
+3. You see the constructor, which we also have in a javascript class. A Constructor is not obligatory in a class component, but most likely you will encounter one, because it gets quite important when we get to concepts like inheritance and state. So to get used to it, it's already included here. In React you pass "props" as an argument to the constructor and also to the `super()` call which has to be called in any constructor. You will not learn about props here, we discuss this in the next lesson, but in short, props are used in react to pass properties from a parent component to child components. But as I said, don't worry too much about that now, we will talk about it in full later on.
```

Comment:
We will need to consider (moving forward) whether or not the individual snippets of code from the originating example should be provided during breakdowns.

At this point the user isn't using their own environment and potentially having to scroll back and forth, which is a minor inconvenience but worth noting.

Reply:
Good point. What we could consider would be showing the snippets again after each point of explanation. Something like:
3. .....

````
constructor() {
        super()
    }
````

I think it is important for the student to see the whole code in one place, especially because they are not used to react by then. However, duplicating code snippets could be good for two reasons:
1. The students won't have to scroll unless they need to see the whole code again
2. The students have another chance of seeing the code, which increases the possibility for them to remember it.